src/messages/proposals.rs

```diff
@@ -182,17 +207,42 @@ impl ProposalQueue {
     }
     /// Returns a new `ProposalQueue` from proposals that were committed and
     /// don't need filtering
-    pub(crate) fn new_from_committed_proposals(
+    pub(crate) fn from_proposals_by_reference(
         ciphersuite: &Ciphersuite,
-        proposals: Vec<MLSPlaintext>,
+        proposals: &'a [&MLSPlaintext],
     ) -> Self {
         let mut proposal_queue = ProposalQueue::new();
         for mls_plaintext in proposals {
-            let queued_proposal = QueuedProposal::new(ciphersuite, &mls_plaintext);
+            let queued_proposal = QueuedProposal::from_mls_plaintext(ciphersuite, &mls_plaintext);
             proposal_queue.add(queued_proposal);
         }
         proposal_queue
     }
+    /// Returns a new `ProposalQueue` from proposals that were committed and
```

Comment:
Not sure if this is the place, but shouldn't we sanity-check that committed proposals don't need to be filtered? Or is that taken care of by #133?

Reply:
Yes, there is no validation yet and we should tackle that in #133.